tests/dragonfly/replication_test.py

```diff
@@ -1836,3 +1836,80 @@ async def test_replicaof_reject_on_load(df_local_factory, df_seeder_factory):
     await c_replica.close()
     master.stop()
     replica.stop()
+
+
+# note: please be careful if you want to change any of the parameters used in this test.
+# changing parameters without extensive testing may easily lead to weak testing case assertion
+# which means eviction may not get triggered.
+@pytest.mark.asyncio
+async def test_policy_based_eviction_propagation(df_local_factory):
+    master = df_local_factory.create(
+        proactor_threads=1, cache_mode="true", maxmemory="256mb", enable_heartbeat_eviction="false"
+    )
+    replica = df_local_factory.create(proactor_threads=1)
+    df_local_factory.start_all([master, replica])
+
+    c_master = master.client()
+    c_replica = replica.client()
+
+    await c_replica.execute_command(f"REPLICAOF localhost {master.port}")
+    await wait_available_async(c_replica)
+
+    pipe = c_master.pipeline(transaction=False)
+    batch_fill_data(client=pipe, gen=gen_test_data(3700000))
+    await pipe.execute()
+
+    info = await c_master.info("stats")
+    assert info["evicted_keys"] > 0, "Weak testcase: policy based eviction was not triggered."
+
+    await check_all_replicas_finished([c_replica], c_master)
+    keys_master = await c_master.execute_command("keys *")
+    keys_replica = await c_replica.execute_command("keys *")
+    assert set(keys_master) == set(keys_replica)
```

Comment:
just reminding to be cautious with the `keys_output_limit` limit, it might be less than there are keys (currently its 8k)

Reply:
got it, thanks!

Reply:
> just reminding to be cautious with the `keys_output_limit` limit, it might be less than there are keys (currently its 8k), nit: the client has a keys() function

Just increased this value to cover the total number of keys that will be used in this test when starting master and replica.

Reply:
When running your code on my laptop it took 8 minutes to run. I am now writing a pytest test useing debug populate which will enter the eviction and will finish fast. Will send you the code